pyro/poutine/lift_messenger.py

```diff
@@ -26,6 +29,22 @@ def __init__(self, prior):
         super(LiftMessenger, self).__init__()
         self.prior = prior
 
+    def __enter__(self):
+        if is_validation_enabled() and isinstance(self.prior, dict):
+            self._param_hits = set()
+            self._param_misses = set()
+        return super(LiftMessenger, self).__enter__()
+
+    def __exit__(self, *args, **kwargs):
+        if is_validation_enabled() and isinstance(self.prior, dict):
+            extra = set(self.prior) - self._param_hits
```

Comment:
note this wont catch the situation in which the user underspecifies the amount of priors for the amount of params.